### PR TITLE
UI Plugin: Added UI unit tests

### PR DESCRIPTION
**Pull request by raymondz1**
## What this PR does / why we need it
add unit tests for the docker provider. 

## Which issue(s) this PR fixes

Fixes: https://github.com/vmware-tanzu/community-edition/issues/4541

### CODE REVIEW

File: cli/cmd/plugin/ui/web/tanzu-ui/src/shared/components/management-cluster-settings-step/ManagementClusterSettings.test.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+// App imports
+
+describe('ManagementClusterSettings', () => {
+    it('should render', async () => {
+        const view = render(<ManagementClusterSettings />);
+        expect(view).toBeDefined();
+    });
+
+    it('show a message if there is only one option for the control plane node profile', () => {
```
Comment (swalner-vmware) [inline cli/cmd/plugin/ui/web/tanzu-ui/src/shared/components/management-cluster-settings-step/ManagementClusterSettings.test.tsx:17]: nit: shouldn't this title be more generic: 'show a message if one is provided'?

File: cli/cmd/plugin/ui/web/tanzu-ui/src/shared/components/management-cluster-settings-step/ManagementClusterSettings.tsx [hunk 2]
```diff
@@ -63,7 +63,7 @@ function ManagementClusterSettings(props: Partial<MCSettings>) {
         setSelectedProfile(event.target.value);
     };
 
-    const handleClusterNameChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const handleClusterNameChange = (event: any) => {
```
Comment (swalner-vmware) [inline cli/cmd/plugin/ui/web/tanzu-ui/src/shared/components/management-cluster-settings-step/ManagementClusterSettings.tsx:66]: I'm curious why we would abandon the type specification here?

File: cli/cmd/plugin/ui/web/tanzu-ui/src/shared/components/management-cluster-settings-step/ManagementClusterSettings.tsx [hunk 3]
```diff
@@ -106,7 +107,7 @@ function ManagementClusterSettings(props: Partial<MCSettings>) {
                             <label>Select a control plane node profile</label>
                             {nodeProfiles.map((nodeProfile, index) => {
                                 return (
-                                    <CdsRadio cds-layout="m:lg m-l:xl p-b:sm" key={index}>
+                                    <CdsRadio cds-layout="m:lg m-l:xl p-b:sm" key={index} data-testid="cds-radio">
```
Comment (swalner-vmware) [inline cli/cmd/plugin/ui/web/tanzu-ui/src/shared/components/management-cluster-settings-step/ManagementClusterSettings.tsx:110]: I'm guessing it's my oversight, but I'm not seeing where this `data-testid` is being used

**Review by swalner-vmware — APPROVED** [review]

File: cli/cmd/plugin/ui/web/tanzu-ui/src/shared/components/management-cluster-settings-step/ManagementClusterSettings.test.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
+// React imports
+import React from 'react';
+
+// Library imports
+import { render, waitFor, screen, fireEvent } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import ManagementClusterSettings from './ManagementClusterSettings';
```
Comment (miclettej) [inline cli/cmd/plugin/ui/web/tanzu-ui/src/shared/components/management-cluster-settings-step/ManagementClusterSettings.test.tsx:7]: move to App imports

File: cli/cmd/plugin/ui/web/tanzu-ui/src/shared/components/management-cluster-settings-step/ManagementClusterSettings.test.tsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        const handleValueChangeMock = jest.fn();
+        render(<ManagementClusterSettings handleValueChange={handleValueChangeMock} />);
+        const input = screen.getByLabelText('cluster name');
+        fireEvent.change(input, { target: { value: 'myTestCluster' } });
+        expect(handleValueChangeMock).toHaveBeenCalled();
+    });
+
+    it('select the third option in the control plane profile', async () => {
+        const user = userEvent.setup();
+
```
Comment (miclettej) [inline cli/cmd/plugin/ui/web/tanzu-ui/src/shared/components/management-cluster-settings-step/ManagementClusterSettings.test.tsx:41]: nitpick remove line break

**Review by miclettej — APPROVED** [review]

File: cli/cmd/plugin/ui/web/tanzu-ui/src/shared/components/management-cluster-settings-step/ManagementClusterSettings.test.tsx (hunk 1 above)
Comment (raymondz1) [reply cli/cmd/plugin/ui/web/tanzu-ui/src/shared/components/management-cluster-settings-step/ManagementClusterSettings.test.tsx]: fixed.

File: cli/cmd/plugin/ui/web/tanzu-ui/src/shared/components/management-cluster-settings-step/ManagementClusterSettings.tsx (hunk 2 above)
Comment (raymondz1) [reply cli/cmd/plugin/ui/web/tanzu-ui/src/shared/components/management-cluster-settings-step/ManagementClusterSettings.tsx]: good catch. I was trying something else, but didn't change it back.

File: cli/cmd/plugin/ui/web/tanzu-ui/src/shared/components/management-cluster-settings-step/ManagementClusterSettings.tsx (hunk 3 above)
Comment (raymondz1) [reply cli/cmd/plugin/ui/web/tanzu-ui/src/shared/components/management-cluster-settings-step/ManagementClusterSettings.tsx:110]: we need it to get the element in the unit test.

File: cli/cmd/plugin/ui/web/tanzu-ui/src/shared/components/management-cluster-settings-step/ManagementClusterSettings.test.tsx (hunk 5 above)
Comment (raymondz1) [reply cli/cmd/plugin/ui/web/tanzu-ui/src/shared/components/management-cluster-settings-step/ManagementClusterSettings.test.tsx]: removed

**Pull request merged**